src/Generator/Passes/ParamTypeToInterfacePass.cs

```diff
@@ -7,20 +7,22 @@ public class ParamTypeToInterfacePass : TranslationUnitPass
     {
         public override bool VisitFunctionDecl(Function function)
         {
-            if (!function.IsOperator)
-            {
+            if (!function.IsOperator || function.Parameters.Count > 1)
                 ChangeToInterfaceType(function.ReturnType);
-                foreach (Parameter parameter in function.Parameters)
-                {
-                    ChangeToInterfaceType(parameter.QualifiedType);
-                }
-            }
+
+            if(function.OperatorKind != CXXOperatorKind.Conversion &&
+                                    function.OperatorKind != CXXOperatorKind.ExplicitConversion)
+                     foreach (var parameter in function.Parameters)
```

Comment:
Sorry, i copied few changes between branches and that part didn't got copied. I have corrected it now.